pkg/media/pipe.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This is a copy of io/pipe.go adapted to media.
+
+package media
```

Comment:
Could you comment on the differences with the Go native library implementation?

Reply:
Sure, will add a comment. It's just a generic version of the same code, plus it's adapted to media interfaces in SIP (`WriteSample` instead of `Write`).

Reply:
@biglittlebigben Done!

Reply:
There is always a maintenance question about forking library code. Is this used in the tests only? If so, should we move it to a test related package to discourage use in service code?

Reply:
Only used in testing code for now, but it could be useful for service code as well later.

The whole `media` package is kind of experimental. The goal is to see what interfaces make sense and if it could be used in other services or Go SDK.

The Reader/Writer interfaces are well-known, so I wanted to try using similar approach for reading/writing RTP and/or audio data. For now it's working quite well. So I'm expanding it a bit over time.